src/core/PWCharPool.cpp

```diff
@@ -542,3 +544,40 @@ bool CPasswordCharPool::CheckMasterPassword(const StringX &pwd, StringX &error)
     return false;
   }
 }
+
+double CPasswordCharPool::CalculatePasswordStrength(const StringX &password)
+{
+  if (password.empty()) {
+    return 0.0; // Empty password has no strength
+  }
+
+  // Determine the character pool size
+  std::unordered_set<char> charSet(password.begin(), password.end());
+  int poolSize = 0;
+
+  bool hasLower = false, hasUpper = false, hasDigits = false, hasSpecial = false;
+  for (char c : charSet) {
+    if (std::islower(c)) hasLower = true;
+    else if (std::isupper(c)) hasUpper = true;
+    else if (std::isdigit(c)) hasDigits = true;
+    else hasSpecial = true;
+  }
+
+  if (hasLower) poolSize += 26; // a-z
+  if (hasUpper) poolSize += 26; // A-Z
+  if (hasDigits) poolSize += 10; // 0-9
+  if (hasSpecial) poolSize += 32; // Special characters (approximation)
+
+  // Calculate entropy
+  double entropy = password.size() * std::log2(poolSize);
+
+  // Normalize entropy to a range of 0 to 100
+  double maxEntropy = 100.0; // Threshold for max entropy
+  double normalizedStrength = (entropy / maxEntropy) * 100.0;
```

Comment:
Nit:
```suggestion
  double normalizedStrength = (entropy / maxEntropy) * maxEntropy;
```
Same on next lines

Reply:
I removed this, because I think this is code from experimenting with the entropy calculation. The calculation result can essentially be used directly and doesn't really need to be converted, scaled, or normalized. The return value is limited to 100, which also isn't really necessary. However, this is done with the knowledge that at the UI level, the strength meter uses a value range from 0 to 100. According to various sources, an entropy value of approximately 70 or more already considers a password strong. Therefore, I thought that values ​​greater than 100 no longer provide any added value.